### DISCUSSION TITLE
Chrome extension with Next JS API Call - POST Request net::ERR_FILE_NOT_FOUND

### DISCUSSION THREAD

**Discussion by MarvingMoreton** [vercel/next.js#50798] [Help]
### Summary

Hello, 

I am building a chrome extension with next js. I am at the point where I have migrate my next js chrome extension from a local environment, to an extension environment (migrating localStorage to chrome Storage, etc).

However, it seems that I cannot access my server API call as I keep having the error:

49-74616f766e74a443.js:6 POST chrome-extension://xxxxx/api/prompt net::ERR_FILE_NOT_FOUND

Where my POST request with Axios is failing.

I have check multiples documentation, and opened my api routes through next.config.js:
`const nextConfig = {
  reactStrictMode: true,

  images: {
    unoptimized: true
  },
  async headers() {
    return [
      {
        // matching all API routes
        source: '/api/:path*',
        headers: [
          { key: 'Access-Control-Allow-Credentials', value: 'true' },
          { key: 'Access-Control-Allow-Origin', value: '*' },
          {
            key: 'Access-Control-Allow-Methods',
            value: 'GET,OPTIONS,PATCH,DELETE,POST,PUT'
          },
          {
            key: 'Access-Control-Allow-Headers',
            value:
              'X-CSRF-Token, X-Requested-With, Accept, Accept-Version, Content-Length, Content-MD5, Content-Type, Date, X-Api-Version'
          }
        ]
      }
    ]
  }
}
module.exports = nextConfig
`

The API routes is hosted on Vercel, but it's still not working. Do I need to set up a proxy to access publicly the API routes?
Still new to the full stack world here!

Appreciate the help :D

**Comment by remusris** [comment]
```
import { NextApiRequest, NextApiResponse } from "next";

export default function handler(req: NextApiRequest, res: NextApiResponse) {

  // Set the CORS headers to allow any origin
  res.setHeader("Access-Control-Allow-Origin", "*");
  res.setHeader(
    "Access-Control-Allow-Methods",
    "GET, POST, PUT, DELETE, OPTIONS",
  );
  res.setHeader(
    "Access-Control-Allow-Headers",
    "X-Requested-With, Access-Control-Allow-Origin, X-HTTP-Method-Override, Content-Type, Authorization, Accept",
  );

  // Handle preflight requests (CORS)
  if (req.method === "OPTIONS") {
    return res.status(200).end();
  }

  res.status(200).json({ text: "Hello" });
}
```

This got me a response from the chrome extension.  This is coming from a localhost though.
